### DISCUSSION TITLE
"set-language" for all files with extension ".whatever" or name matching "whatever"

### DISCUSSION THREAD

**Discussion by helmesjo** [helix-editor/helix#4485] [Q&A] [answered]
Hi! Maybe there already is an easy way to do this and I've just missed it.

I'd like to specify in (preferably) my `config.toml` that a specific file extension (or in my case, files named `buildfile`) be highlighted using existing highlighting for another language (`bash` in this case).
Currently I can do this on a _per buffer_ basis (using `set-language bash`), but I'd like helix to do this automatically for any file opened that's named `buildfile` or ends with `.buildfile`. Is this possible?

Thanks!

**Comment by the-mikedavis** [comment, marked as answer]
You can create a custom `languages.toml` config that overrides the file-types for bash to include buildfile https://docs.helix-editor.com/master/languages.html

```toml
# ~/.config/helix/languages.toml
[[language]]
name = "bash"
file-types = ["sh", "bash", "zsh", ".bash_login", ".bash_logout", ".bash_profile", ".bashrc", ".profile", ".zshenv", ".zlogin", ".zlogout", ".zprofile", ".zshrc", "APKBUILD", "PKGBUILD", "eclass", "ebuild", "bazelrc", "buildfile"]

**Reply by helmesjo** [reply]
@the-mikedavis Ah great, thanks! But as you mention, then it _overrides_ so I need to make sure to list all file-types? I can't "append" so to say?

**Reply by the-mikedavis** [reply]
Yep correct

**Reply by helmesjo** [reply]
One note that threw me off a bit: `config-reload` was not enough after adding the `languages.toml` (maybe I should've done regular buffer reload?). I had to restart helix for it to take effect.

**Reply by edrozenberg** [reply]
> I can't "append" so to say?

Same, wish I could add an extension to some lang def. For ex. I want to add the `.csh` extension to `bash` (even if imperfect) because otherwise there's no highlighting for `.csh` files at all. As it is I'd need to copy and modify the existing 50+ line bash definition in https://github.com/helix-editor/helix/blob/master/languages.toml to my own config file. And if the original source definition ever changes, there's no way to keep track of it. Not a great solution so I won't do this, simpler to just manually change the language with `:set-language` every time I open one of these files. It would be far better if I could just say in my own config "use the bash def with anything ending in `.csh`".
